packages/auth/src/authClient.ts

```diff
@@ -245,32 +248,21 @@ export class AuthClient {
   public async loginWithOauth(
     params: LoginWithOauthParams,
   ): Promise<AuthSession> {
-    const { targetPublicKey } = await this.getTekStamper();
-    const oauthConfig = await this.signerHttpClient.request({
-      route: "signer/v1/prepare-oauth",
-      method: "POST",
-      body: { nonce: getOauthNonce(targetPublicKey) },
-    });
-    const authUrl = getOauthProviderUrl({
-      oauthParams:
-        params.mode === "redirect" ? { ...params, redirectUrl: "/" } : params, // TO DO: clean up redirect vs popup path
-      turnkeyPublicKey: targetPublicKey,
-      oauthCallbackUrl: "https://signer.alchemy.com/callback",
-      oauthConfig,
-    });
-    const response = await this.handleOauthFlow(authUrl, params.mode);
-    if (response.status === "SUCCESS") {
-      return this.completeAuthWithBundle({
-        bundle: response.bundle!,
-        orgId: response.orgId!,
-        idToken: response.idToken,
-        authType: "oauth",
-      });
-    } else if (response.status === "ACCOUNT_LINKING_CONFIRMATION_REQUIRED") {
-      // TODO: decide what to do here.
-      throw new Error("Account linking confirmation required");
-    } else {
-      throw new Error(`Unknown OAuth flow response: ${response.status}`);
+    switch (params.mode) {
+      case "redirect":
+        const authUrl = await this.buildAuthUrl(params);
+        const redirectResponse = await this.handleOauthFlow({
+          authUrl,
+          mode: params.mode,
+        });
+        return await this.processOAuthResponse(redirectResponse);
+      default: // popup
+        const authUrlPromise = this.buildAuthUrl(params);
+        const popupResponse = await this.handleOauthFlow({
+          authUrl: authUrlPromise,
+          mode: params.mode,
+        });
+        return await this.processOAuthResponse(popupResponse);
```

Comment:
nit: might be a little simpler to just do this instead of repeating everything in the 2 different branches here. just a suggestion though & i'm fine w/ it as-is if you prefer.

```suggestion
        const authUrlPromise = this.buildAuthUrl(params);
        const redirectResponse = await this.handleOauthFlow({
          authUrl: params.mode === "popup" ? authUrlPromise : await authUrlPromise,
          mode: params.mode,
        });
        return await this.processOAuthResponse(redirectResponse);
```

Reply:
this is way cleaner, I like it